AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -102,10 +108,10 @@ - (instancetype)init {
 #pragma mark - MSServiceInternal
 
 + (instancetype)sharedInstance {
-  static id sharedInstance = nil;
-  static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
-    sharedInstance = [[MSDistribute alloc] init];
+    if (sharedInstance == nil) {
```

Comment:
No need for `if` here

Reply:
We seem to have an `if` here in other modules. Why don't we need one here?

Reply:
Because of `dispatch_once` usage

Reply:
The dispatch once will make sure there's only one assignment across calls and threads. It was probably overlooked in other places.